src/Tools/dotnet-format/CodeFormatter.cs

```diff
@@ -94,52 +97,67 @@ private static async Task<int> FormatFilesInSolutionAsync(ILogger logger, Soluti
 
                 logger.LogInformation(Resources.Formatting_code_files_in_project_0, project.Name);
 
-                formattedSolution = await FormatFilesInProjectAsync(logger, project, codingConventionsManager, optionsApplier, cancellationToken).ConfigureAwait(false);
-            }
-
-            if (!solution.Workspace.TryApplyChanges(formattedSolution))
-            {
-                logger.LogError(Resources.Failed_to_save_formatting_changes);
-                return 1;
+                var formattedSolution = await FormatFilesInProjectAsync(logger, project, codingConventionsManager, optionsApplier, cancellationToken).ConfigureAwait(false);
+                if (!workspace.TryApplyChanges(formattedSolution))
+                {
+                    logger.LogError(Resources.Failed_to_save_formatting_changes);
+                    return 1;
+                }
             }
 
             return 0;
         }
 
         private static async Task<Solution> FormatFilesInProjectAsync(ILogger logger, Project project, ICodingConventionsManager codingConventionsManager, EditorConfigOptionsApplier optionsApplier, CancellationToken cancellationToken)
         {
-            var formattedSolution = project.Solution;
-
             var isCommentTrivia = project.Language == LanguageNames.CSharp
                 ? IsCSharpCommentTrivia
                 : IsVisualBasicCommentTrivia;
 
+            var formattedDocuments = new List<(DocumentId documentId, Task<SourceText> formatTask)>();
             foreach (var documentId in project.DocumentIds)
             {
-                var document = formattedSolution.GetDocument(documentId);
+                var document = project.Solution.GetDocument(documentId);
                 if (!document.SupportsSyntaxTree)
                 {
                     continue;
                 }
 
-                var syntaxTree = await document.GetSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
-                if (GeneratedCodeUtilities.IsGeneratedCode(syntaxTree, isCommentTrivia, cancellationToken))
+                var formatTask = Task.Run(async () =>
                 {
-                    continue;
-                }
+                    var syntaxTree = await document.GetSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
total nit.  consider: extracting out the logic here into a local function or sibling method.  i personally like the task-management and solution-application being separated from the actual job of formatting.  But it's entirely a personal preference.  This works equally as well.